src/lib/models/useModelGist.ts

```diff
@@ -0,0 +1,160 @@
+type ResourceQuery = Query[number]
+type GistResourceQuery = Omit<ResourceQuery, 'resource'> & {
+    resource: GistResourceString
+}
+
+// note that we do not support parallel queries
+// this makes it significantly easier to implement
+// and parallel use of gist queries shouldn't be a common use case
+type GistQuery = {
+    result: GistResourceQuery
+}
+
+type GistQueryResult<Response> = {
+    result: Response
+}
+
+function createGistQuery(
+    resource: GistResourceString,
+    params?: GistParams
+): GistQuery {
+    return {
+        result: {
+            resource: `${resource}`,
+            params: ({ page, ...dynamicParams }) => ({
+                ...params,
+                ...dynamicParams,
+                page,
+            }),
+        },
+    }
+}
+
+function usePagination(
+    refetch: QueryResponse['refetch'],
+    data?: unknown
+): GistPaginator | null {
+    let pager: GistPager | undefined
+
+    if (isDataCollection(data)) {
+        pager = data.pager
+    }
+    const getNextPage = useCallback(() => {
+        if (!pager?.nextPage) {
+            return false
+        }
+        refetch({ page: pager.page + 1 })
+        return true
+    }, [refetch, pager])
+
+    const getPrevPage = useCallback(() => {
+        if (!pager?.prevPage) {
+            return false
+        }
+        refetch({ page: pager.page - 1 })
+        return true
+    }, [refetch, pager])
+
+    return {
+        getNextPage,
+        getPrevPage,
+    }
+}
+
+type GistPaginator = {
+    getNextPage: () => boolean
+    getPrevPage: () => boolean
+}
+
+type BaseUseModelGistResult<Response> = Pick<
+    QueryResponse,
+    'loading' | 'error' | 'called' | 'refetch'
+> & {
+    data?: Response
+}
+
+type UseModelGistResultPaginated<Response> =
+    BaseUseModelGistResult<Response> & {
+        pagination: GistPaginator
+    }
+type UseModelGistResult<Response extends GistResponse> =
+    | BaseUseModelGistResult<Response>
+    | UseModelGistResultPaginated<Response>
+
+const isDataCollection = (
+    data: unknown
+): data is GistCollectionResponse<IdentifiableObject, GistResourceString> => {
+    // gist endpoints are always paged if they're collections
+    return (data as GistCollectionResponse)?.pager !== undefined
+}
+
+export function useModelGist<Response extends GistPagedResponse>(
+    gistResource: GistResourceString,
+    params?: GistParams
+): UseModelGistResultPaginated<Response>
+
+export function useModelGist<Response extends GistObjectResponse>(
+    gistResource: GistResourceString,
+    params?: GistParams
+): UseModelGistResult<Response>
+
+export function useModelGist<Response extends GistResponse>(
+    gistResource: GistResourceString,
+    params?: GistParams
+): UseModelGistResult<Response> {
+    const [gistQuery] = useState<GistQuery>(
+        createGistQuery(gistResource, params)
+    )
+    // workaround to keep previous data while fetching, to prevent flickering
+    // "mimicks" react-query's 'keepPreviousData'
```

Comment:
I think we should expose this from app-runtime. But since we're going to have a bigger discussion about the future of `useDataQuery` and our data solution - I decided to implement a workaround here.